controllers/playlist.controller.js

```diff
@@ -0,0 +1,144 @@
+const { extend } = require("lodash");
+const { Playlist } = require("../models/Playlist.model");
+const getAllPlaylists = async (req, res) => {
+  try {
+    const allPlaylists = await Playlist.find({}).populate("videos");
+    res.status(200).json({
+      success: true,
+      playlists: allPlaylists,
+    });
+  } catch (err) {
+    res.status(400).json({
+      success: false,
+      message: "Error in getting playlists",
+      errMessage: err.errMessage,
+    });
+  }
+};
+const addAPlaylist = async (req, res) => {
+  const playlistData = req.body;
+  const newPlaylist = new Playlist({ ...playlistData, videos: [] });
+  try {
+    const savedPlaylist = await newPlaylist.save();
+    res.status(201).json({
+      success: true,
+      playlist: savedPlaylist,
+    });
+  } catch (err) {
+    res.status(400).json({
+      success: false,
+      message: "Error in adding a playlist",
+      errMessage: err.errMessage,
+    });
+  }
+};
+const getPlaylistDetails = (req, res) => {
+  const playlist = req.playlist;
+  res.status(200).json({
+    success: true,
+    playlist,
+  });
+};
+const updatePlaylist = async (req, res) => {
+  const updatePlaylistDetails = req.body;
+  let playlist = req.playlist;
+  playlist = extend(playlist, updatePlaylistDetails);
+  try {
+    const updatedPlaylist = playlist.save();
+    res.status(201).json({
+      success: true,
+      message: "Updated playlist",
+      playlist: updatedPlaylist,
+    });
+  } catch (err) {
+    res.status(400).json({
+      success: false,
+      message: "error in updating playlist details",
+      errMessage: err.errMessage,
+    });
+  }
+};
+const removePlaylist = async (req, res) => {
+  try {
+    const playlist = req.playlist;
+    const deletedPlaylist = await playlist.remove();
+    res.status(200).json({
+      success: true,
+      message: "deleted playlist",
+      playlist: deletedPlaylist,
+    });
+  } catch (err) {
+    res.status(400).json({
+      success: false,
+      message: "Error in deleting playlist",
+      playlist,
+    });
+  }
+};
+const getPlaylistVideos = async (req, res) => {
+  const playlistId = req.playlist._id;
+  try {
+    const playlistVideosData = await Playlist.findById(playlistId)
+      .populate("videos")
+      .select("__v");
```

Comment:
If you want to remove __v from data then add a - sign before that e.g., `.select("-__v")`